letta/schemas/llm_config.py

```diff
@@ -17,7 +17,9 @@ class LLMConfig(BaseModel):
 
     # TODO: 🤮 don't default to a vendor! bug city!
     model: str = Field(..., description="LLM model name. ")
-    model_endpoint_type: str = Field(..., description="The endpoint type for the model.")
+    model_endpoint_type: Literal["openai", "anthropic", "cohere", "google_ai", "azure", "groq", "ollama"] = Field(
```

Comment:
Do we also want to support `local` in this literal? Since I think `local` is being used elsewhere in the code, also other legacy types like `webui`, `vllm`, etc (https://github.com/cpacker/MemGPT/blob/6d154bfcbd16bbeeb5fccf71cb59b9ecbe77407d/paper_experiments/utils.py#L31)

Maybe the best thing for now is to leave this as a non-literal string, but once we eventually test all legacy endpoints, revert?

Reply:
(I do generally agree that this should be a `Literal`, not a `str`, I'm just worried about breaking people old setups that are hooked up to eg LMStudio or local vLLM using old `model_endpoint_type` types

Reply:
Yeah, I can add other endpoint types, but I think migrating now is okay - I'll keep an eye on if people complain, and even if they do I think it's an pretty transparent fix on their end (just adding whatever key they're missing to the Literal).

Reply:
Adding:

```
"webui", "webui-legacy", "lmstudio", "lmstudio-legacy", "llamacpp", "koboldcpp", "vllm", "hugging-face"
```